src/grid_map_geo.cpp

```diff
@@ -99,7 +100,7 @@ bool GridMapGeo::initializeFromGeotiff(const std::string &path) {
 
   double mapcenter_e = originX + pixelSizeX * width * 0.5;
   double mapcenter_n = originY + pixelSizeY * height * 0.5;
-  maporigin_.espg = ESPG::CH1903_LV03;
+  maporigin_.espg = static_cast<ESPG>(std::stoi(epsg_code));
```

Comment:
FYI, `std::stoi` could throw if the `OGRSpatialReference` doesn't have the `AUTHORITY` attribute because it will return null. You might want to wrap it in try-catch, or return early if the `epsg_code == nullptr`